backend/simplemap/simplemap.go

```diff
@@ -0,0 +1,44 @@
+package simplemap
+
+import (
+	"github.com/heetch/confita"
+	"github.com/pkg/errors"
+	"reflect"
+
+	"context"
+)
+
+// Backend that loads config from map
+type Backend struct {
+	theMap map[string]interface{}
+}
+
+// NewBackend creates a simplemap backend.
+func NewBackend(theMap map[string]interface{}) *Backend {
+	return &Backend{
+		theMap: theMap,
+	}
+}
+
+// LoadStruct takes a struct config and loads the map into it
+func (b *Backend) LoadStruct(ctx context.Context, cfg *confita.StructConfig) error {
+	for _, f := range cfg.Fields {
+		mapVal := b.theMap[f.Key]
+		if mapVal == nil {
+			continue
+		}
+		mapRef := reflect.ValueOf(mapVal)
+		f.Value.Set(mapRef)
```

Comment:
This will panic if the `mapVal` does not have the same type as `f.Value`. We could return an error if the types aren't identical, but this might be perceived as being overly restrictive, because (for example) you wouldn't be able to use an int-valued map item to fill a float-valued struct field.

Perhaps it would be easiest to implement the Backend on `map[string]string` instead of `map[string]interface{}` and then just implement the `Get` method?